client/reader/list-manage/list-form.jsx

```diff
@@ -60,17 +62,17 @@ export default function ListForm( { isCreateForm, isSubmissionDisabled, list, on
 
 			{ ! isCreateForm && (
 				<FormFieldset>
-					<FormLabel htmlFor="list-slug">{ translate( 'Slug (Required)' ) }</FormLabel>
+					<FormLabel htmlFor="list-slug">{ translate( 'Slug' ) }</FormLabel>
 					<FormTextInput
 						data-key="slug"
+						// NOTE: Slug modification currently doesn't work in the API.
+						disabled
 						id="list-slug"
-						isValid={ isSlugValid }
 						name="list-slug"
-						onChange={ onChange }
 						value={ formList.slug }
 					/>
 					<FormSettingExplanation>
-						{ translate( 'The slug for the list. This is used to build the URL to the list.' ) }
+						{ translate( 'This is used to build the URL to the list.' ) }
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.